File: baekjoon/1차원 배열/8958.js
const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

let input = []

rl.on('line', (line) => {
  input.push(line)
}).on('close', () => {
  const T = parseInt(input[0])
  let result = ''
  for (i = 1; i <= T; i++) {
    let score = 0
    let sum = 0
    for (j = 0; j < input[i].length; j++) {
      if (input[i][j] === 'O') {
        score = score + sum + 1
        sum++
      } else {
        sum = 0
      }
    }
    result = result + `${score}\n`
  }
  console.log(result)
  process.exit()
})

File: baekjoon/단계별로 풀어보기/1차원 배열/8958.py
T = int(input())
result = ""
for i in range(T):
  answer = input()
  score = 0
  acc = 0
  for j in answer:
    if(j=="O"):
      score = score + acc + 1
      acc = acc + 1
    else:
      acc = 0
  result = result + f"{score}\n"
print(result)